### PR TITLE
jwt 통한 회원 인증 및 인가 

### LINKED ISSUE #3: jwt 사용한 회원 인증
## Description

> spring security와 jwt 활용한 회원 인증 및 인가 구현

## Progress

- [x] 회원 로그인 성공 시 access token , refresh token 발급
- [x] refresh token은 redis에 저장
- [x] access token 만료 시 refresh token 통해 재발급
- [x] 회원 로그아웃 시 redis에서 refresh token 삭제
- [x] 회원 로그아웃 시  access token redis에 blackList 저장
- [x] 예외 처리
- [ ] 테스트 코드 (생각 좀만 더해보고...)

### PR DESCRIPTION
**Pull request by EunjungOh123**

### CODE REVIEW

**Review by younghoondoodoom — COMMENTED** [review]
정말 고생 하셨습니다! 이해하면서 보는데에만 해도 시간 꽤 걸렸는데 고민 엄청 많이 하셨을 것 같아요. 제 생각 코멘트 남겼으니 참고 부탁드립니다! 그리고 커밋을 최대한 잘게 나눠주셔서 코드 보기가 편했습니다. 감사합니다!

File: src/main/java/com/gaethering/gaetheringserver/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        DaoAuthenticationProvider authProvider = new DaoAuthenticationProvider();
+
+        authProvider.setUserDetailsService(userDetailsService);
+        authProvider.setPasswordEncoder(passwordEncoder());
+
+        return authProvider;
+    }
+
+    @Bean
+    public BCryptPasswordEncoder passwordEncoder() { // 비밀번호 암호화
```
[the remark this answers is not in the document]
Comment (EunjungOh123) [reply src/main/java/com/gaethering/gaetheringserver/config/SecurityConfig.java]: PasswordEncoder 빈 등록 수정하는 과정에서 주석은 삭제했습니다!

File: src/main/java/com/gaethering/gaetheringserver/filter/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtProvider jwtProvider;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+
+        if (!request.getRequestURI().contains("/login")
+                && !request.getRequestURI().contains("/exception")) {
```
[the remark this answers is not in the document]
Comment (EunjungOh123) [reply src/main/java/com/gaethering/gaetheringserver/filter/JwtAuthenticationFilter.java:27]: JwtAuthenticationFilter 같은 경우 사용자가 정의한 필터에서 인증 및 권한 작업을 진행하기 때문에 Spring Security의 UsernamePasswordAuthenticationFilter 이전에 등록하는 설정이 되어있어 별개로 설정해야 하는 걸로 알고 있습니다! if문 사용하기 보다는 filter에서 제외하는 메서드를 사용하는 것이 나을 거 같아서 찾아보고 수정하도록 하겠습니다!

File: src/main/java/com/gaethering/gaetheringserver/util/CustomAccessDeniedHandler.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Slf4j
+@Component
+public class CustomAccessDeniedHandler implements AccessDeniedHandler {
+
+    @Override
+    public void handle(HttpServletRequest request,
+                       HttpServletResponse response, AccessDeniedException accessDeniedException)
+            throws IOException {
+
+        log.error("UnAuthorized!!! " + accessDeniedException.getMessage());
```
[the remark this answers is not in the document]
Comment (EunjungOh123) [reply src/main/java/com/gaethering/gaetheringserver/util/CustomAccessDeniedHandler.java]: 로그 쪽은 info 로 수정했습니다!

File: src/main/java/com/gaethering/gaetheringserver/member/exception/controller/MemberAuthExceptionController.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    @GetMapping("/accessDenied")
+    public String accessDenied () {
+        throw new AuthorizationFailException(MemberErrorCode.FAIL_TO_AUTHORIZATION);
+    }
+
+    @GetMapping("/authenticationFailed")
+    public String authenticationFailed () {
+        throw new AuthenticationFailException(MemberErrorCode.FAIL_TO_AUTHENTICATION);
+    }
+}
```
[the remark this answers is not in the document]
Comment (EunjungOh123) [reply src/main/java/com/gaethering/gaetheringserver/member/exception/controller/MemberAuthExceptionController.java:23]: 인증과 권한 관련 에러가 발생하면 각각 CustomAuthenticationEntryPoint, CustomAccessDeniedHandler 클래스로 오게 되는데 여기서 에러를 던지게 되어도 GlobalExceptionHandler에서 처리하지 못해서 이런 식으로 생각해봤는데 조금 더 생각해봐야 할거 같습니다!

**Comment by naekang** [comment]
|File|Coverage [39.78%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/2a69d25e22be49394eafc9f9fd7dc082102bdc57/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|42.33%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [39.78%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/80982533a18c4d625d272ca5eb9e661956f4b136/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|42.33%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [32.45%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[Category.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FCategory.java)|10.81%|:x:|
|[Post.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FPost.java)|9.33%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/9fa46302ec48f86b2cd34e530030c84f1d86252a/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|43.88%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [32.45%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[Category.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FCategory.java)|10.81%|:x:|
|[Post.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FPost.java)|9.33%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|43.88%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [32.45%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[Category.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FCategory.java)|10.81%|:x:|
|[Post.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FPost.java)|9.33%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|43.88%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [32.45%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[Category.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FCategory.java)|10.81%|:x:|
|[Post.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FPost.java)|9.33%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|43.88%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [32.45%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[Category.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FCategory.java)|10.81%|:x:|
|[Post.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FPost.java)|9.33%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|43.88%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [32.45%]|:x:|
|:-|:-:|:-:|
|[MemberErrorCode.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fexception%2Ferrorcode%2FMemberErrorCode.java)|100%|:green_apple:|
|[CustomAccessDeniedHandler.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAccessDeniedHandler.java)|43.75%|:x:|
|[CustomAuthenticationEntryPoint.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FCustomAuthenticationEntryPoint.java)|43.75%|:x:|
|[MemberServiceImpl.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FMemberServiceImpl.java)|39.44%|:x:|
|[RedisUtil.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FRedisUtil.java)|16.36%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Ffilter%2FJwtAuthenticationFilter.java)|14.63%|:x:|
|[CustomUserDetailsService.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fmember%2Fservice%2FCustomUserDetailsService.java)|12%|:x:|
|[Category.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FCategory.java)|10.81%|:x:|
|[Post.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Fboard%2Fdomain%2FPost.java)|9.33%|:x:|
|[JwtProvider.java](https://github.com/Gaethering/Gaethering-Server/blob/db3859e576dde00d2fe3792853e4129f54080aaa/src%2Fmain%2Fjava%2Fcom%2Fgaethering%2Fgaetheringserver%2Futil%2FJwtProvider.java)|8.5%|:x:|

|Total Project Coverage|43.88%|:x:|
|:-|:-:|:-:|

**Pull request closed without merging**